plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftCastPushdown.java

```diff
@@ -393,6 +433,33 @@ protected List<CastTestCase> supportedCastTypePushdown()
                 new CastTestCase("c_numeric_30_2", "integer", "c_integer"),
                 new CastTestCase("c_decimal_negative", "integer", "c_integer"),
 
+                new CastTestCase("c_char_50", "char(10)", "c_char_10"),
+                new CastTestCase("c_bpchar", "char(10)", "c_char_10"),
+
+                new CastTestCase("c_char", "char(1)", "c_nchar"),
+                new CastTestCase("c_char_10", "char(50)", "c_char_50"),
+                new CastTestCase("c_char_10", "char(256)", "c_bpchar"),
+                new CastTestCase("c_char", "char(4096)", "c_char_4096"),
+
+                new CastTestCase("c_varchar_50", "varchar(10)", "c_varchar_10"),
+                new CastTestCase("c_nvarchar_50", "varchar(10)", "c_nvarchar_10"),
+                new CastTestCase("c_text", "varchar(10)", "c_varchar_10"),
+
+                new CastTestCase("c_nvarchar_10", "varchar(50)", "c_nvarchar_50"),
+                new CastTestCase("c_varchar_10", "varchar(50)", "c_text"),
+                new CastTestCase("c_varchar_10", "varchar(50)", "c_varchar_50"),
+                new CastTestCase("c_varchar_10", "varchar(65535)", "c_varchar_50"),
+                new CastTestCase("c_varchar_10", "varchar(256)", "c_nvarchar"),
+                new CastTestCase("c_varchar_10", "varchar(10)", "c_nvarchar_10"),
+                new CastTestCase("c_varchar_10", "varchar(50)", "c_nvarchar_50"),
+                new CastTestCase("c_varchar_10", "varchar(65535)", "c_nvarchar_65535"),
```

Comment:
In join when `castType` and `targetColumn`'s type is different then there will an additional implicit cast, which I think will be redundant test coverage. In this let's focus on explicit cast.

Can we arrange the test cases in some order? This way it will easy to read. Like what we have done for oracle. For casting to `varchar` we could use `c_varchar_65535`.

https://github.com/trinodb/trino/blob/7a3f815d82b4f2ff144361f4edce739bb7f0bbaf/plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleCastPushdown.java#L330-L368